src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -216,32 +248,35 @@ private Service getServiceForFieldDefinition(GraphQLFieldDefinition fieldDefinit
         return info.getService();
     }
 
-    private CompletableFuture<RootExecutionResultNode> callService(ExecutionContext executionContext,
-                                                                   QueryTransformationResult queryTransformerResult,
-                                                                   Service service,
-                                                                   ExecutionStepInfo rootExecutionStepInfo,
-                                                                   Operation operation) {
+    private ServiceExecutionParameters buildServiceExecutionParameters(ExecutionContext executionContext, QueryTransformationResult queryTransformerResult) {
 
-        Document query = queryTransformerResult.getDocument();
-        Map<Field, FieldTransformation> transformationByResultField = queryTransformerResult.getTransformationByResultField();
-        List<MergedField> transformedMergedFields = queryTransformerResult.getTransformedMergedFields();
+        // only pass down variables that are referenced in the transformed query
+        Map<String, Object> contextVariables = executionContext.getVariables();
+        Map<String, Object> variables = new LinkedHashMap<>();
+        for (String referencedVariable : queryTransformerResult.getReferencedVariables()) {
```

Comment:
This logic should live in OverallQueryTransformer IMHO (so that QueryTransformationResult provides everything that is needed to execute query)